src/nvim/CMakeLists.txt

```diff
@@ -252,6 +268,17 @@ add_custom_command(OUTPUT ${GENERATED_OPTIONS}
   DEPENDS ${OPTIONS_GENERATOR} ${OPTIONS_LIST_FILE}
 )
 
+# Make NEOVIM_GENERATED_FOR_SOURCES and NEOVIM_GENERATED_FOR_HEADERS mutually exclusive
+# If a header in NEOVIM_GENERATED_FOR_HEADERS is also included in NEOVIM_GENERATED_SOURCES,
+# the header should be included only in NEOVIM_GENERATED_HEADERS
+
+foreach(hfile ${NEOVIM_GENERATED_FOR_HEADERS})
+  list(FIND NEOVIM_GENERATED_FOR_SOURCES ${hfile} hfile_idx)
+  if(NOT ${hfile_idx} EQUAL -1)
+    list(REMOVE_AT NEOVIM_GENERATED_FOR_SOURCES ${hfile_idx})
```

Comment:
@ZyX-I  I think that there is an implicit premise that files in NEOVIM_GENERATED_FOR_HEADERS are for declaring global variables without extern, and they might not be include-guarded.
I checked the generated files, and got to know that some generated headers are include-guarded and other are not (e.g. ex_cmds_defs.generated.h), and that *.h.generated.h are declaring global variables without extern.
The premise could not be noticed in CMakeLists.txt at sight.
I think if a header is include-guarded, and used for just declaring variables with extern, and functions like stddef.h, it should be able to be included in both of headers and sources.

But anyway, I would not like to expand the arguments to the implicit premise in neovim in this PR, I will fix the comment and manifest the invariance that headers and sources should be mutually exclusive.